From 4e0d557a557ecb89ea24023c1d7ffef53aacfb7c Mon Sep 17 00:00:00 2001
From: Robert Lehmann <robert.lehmann@sitec-systems.de>
Date: Thu, 7 Jan 2016 11:48:29 +0100
Subject: [PATCH 8/9] S Core: Add support for CAN pins to S Core Eval dts

The pinmux and initialisation for flexcan1 and flexcan 2 is added to the
kernel. The neccessary control gpios are altough added to the DT. These
gpios are responsible for controlling the external trancievers and the
termination resistors.

Ticket #880
---
 arch/arm/boot/dts/imx6sx-score.dts | 210 ++++++++++++++++++++++++-------------
 1 file changed, 137 insertions(+), 73 deletions(-)

diff --git a/arch/arm/boot/dts/imx6sx-score.dts b/arch/arm/boot/dts/imx6sx-score.dts
index 051851e..f7b8e06 100644
--- a/arch/arm/boot/dts/imx6sx-score.dts
+++ b/arch/arm/boot/dts/imx6sx-score.dts
@@ -1,5 +1,6 @@
 /*
  * Copyright (C) 2014-2015 Freescale Semiconductor, Inc.
+ * Copyright (C) 2015, 2016 Robert Lehmann <robert.lehmann@sitec-systems.de>
  *
  * This program is free software; you can redistribute it and/or modify
  * it under the terms of the GNU General Public License version 2 as
@@ -13,16 +14,16 @@
 #include "imx6sx-score.dtsi"
 
 / {
-	model = "sitec systems S Core Eval-Board";
-	compatible = "fsl,imx6sx-score", "fsl,imx6sx";
+    model = "sitec systems S Core Eval-Board";
+    compatible = "fsl,imx6sx-score", "fsl,imx6sx";
 
-	chosen {
-		stdout-path = &uart1;
-	};
+    chosen {
+        stdout-path = &uart1;
+    };
 
-	memory {
-		reg = <0x80000000 0x20000000>;
-	};
+    memory {
+        reg = <0x80000000 0x20000000>;
+    };
 
     leds {
         compatible = "gpio-leds";
@@ -31,37 +32,37 @@
             gpios = <&gpio3 19 GPIO_ACTIVE_HIGH>;
             linux,default-trigger = "heartbeat";
         };
+        led {
+            label = "Red_led";
+            gpios = <&gpio3 20 GPIO_ACTIVE_HIGH>;
+        };
     };
 
-	regulators {
-		compatible = "simple-bus";
-		#address-cells = <1>;
-		#size-cells = <0>;
-	};
 };
 
 &cpu0 {
-	operating-points = <
-		/* kHz    uV */
-		996000  1250000
-		792000  1175000
-		396000  1175000
-		198000	1175000
-		>;
-	fsl,soc-operating-points = <
-		/* ARM kHz      SOC uV */
-		996000	1250000
-		792000	1175000
-		396000	1175000
-		198000	1175000
-	>;
+    operating-points = <
+        /* kHz    uV */
+        996000  1250000
+        792000  1175000
+        396000  1175000
+        198000	1175000
+        >;
+    fsl,soc-operating-points = <
+        /* ARM kHz      SOC uV */
+        996000	1250000
+        792000	1175000
+        396000	1175000
+        198000	1175000
+        >;
     arm-supply = <&reg_arm>;
     soc-supply = <&reg_soc>;
 };
 
 &gpc {
-	/* use ldo-bypass, u-boot will check it and configure */
-	fsl,ldo-bypass = <0>;
+    /* use ldo-bypass, u-boot will check it and configure */
+    fsl,ldo-bypass = <0>;
+    fsl,wdog-reset = <1>; /* watchdog select of reset source */
 };
 
 &uart1 {
@@ -70,52 +71,115 @@
     status = "okay";
 };
 
-&usdhc3 {
-	pinctrl-names = "default";
-	pinctrl-0 = <&pinctrl_usdhc3>;
-	bus-width = <4>;
-	cd-gpios = <&gpio7 6 GPIO_ACTIVE_HIGH>;
-	wp-gpios = <&gpio7 7 GPIO_ACTIVE_HIGH>;
-	status = "okay";
+&flexcan1 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_flexcan1>;
+    status = "okay";
 };
 
-&iomuxc {
-	pinctrl-names = "default";
-    pinctrl-0 = <&pinctrl_gpio>;
-
-	imx6x-score {
-        pinctrl_gpio: gpiogrp {
-            fsl,pins = <
-                MX6SX_PAD_LCD1_DATA18__GPIO3_IO_19  0x17059
-                MX6SX_PAD_LCD1_DATA19__GPIO3_IO_20  0x17059
-            >;
-        };
+&flexcan2 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_flexcan2>;
+    status = "okay";
+};
 
-        pinctrl_i2c1: i2c1grp {
-            fsl,pins = <
-                MX6SX_PAD_GPIO1_IO01__I2C1_SDA 0x4001b8b1
-                MX6SX_PAD_GPIO1_IO00__I2C1_SCL 0x4001b8b1
-            >;
-        };
+&usdhc3 {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_usdhc3>;
+    bus-width = <4>;
+    cd-gpios = <&gpio7 6 GPIO_ACTIVE_HIGH>;
+    wp-gpios = <&gpio7 7 GPIO_ACTIVE_HIGH>;
+    status = "okay";
+};
 
-		pinctrl_uart1: uart1grp {
-			fsl,pins = <
-				MX6SX_PAD_GPIO1_IO04__UART1_TX		0x1b0b1
-				MX6SX_PAD_GPIO1_IO05__UART1_RX		0x1b0b1
-			>;
-		};
-
-		pinctrl_usdhc3: usdhc3grp {
-			fsl,pins = <
-				MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x17069
-				MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x10071
-				MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x17069
-				MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x17069
-				MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x17069
-				MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x17069
-				MX6SX_PAD_SD3_DATA4__GPIO7_IO_6		0x17059 /* CD */
-				MX6SX_PAD_SD3_DATA5__GPIO7_IO_7		0x17059 /* WP */
-			>;
-		};
-	};
+&gpmi {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_gpmi_nand_1>;
+    status = "okay";
+    nand-on-flash-bbt;
+};
+
+&iomuxc {
+    pinctrl-names = "default";
+    pinctrl-0 = <&pinctrl_hog>;
+
+    imx6x-score {
+pinctrl_hog: hoggrp {
+                 fsl,pins = <
+                     /* CAN1_EN */
+                     MX6SX_PAD_KEY_ROW4__GPIO2_IO_19     0xc9
+                     /* CAN2_EN */
+                     MX6SX_PAD_KEY_COL4__GPIO2_IO_14     0xc9
+                     /* CAN1_TERM */
+                     MX6SX_PAD_LCD1_DATA21__GPIO3_IO_22  0xc9
+                     /* CAN2_TERM */
+                     MX6SX_PAD_GPIO1_IO08__GPIO1_IO_8    0xc9
+                     /* 5P_EN */
+                     MX6SX_PAD_LCD1_DATA17__GPIO3_IO_18  0xc9
+                     >;
+             };
+
+pinctrl_gpmi_nand_1: gpmi-nand-1 {
+                         fsl,pins = <
+                             MX6SX_PAD_NAND_CLE__RAWNAND_CLE         0xb0b1
+                             MX6SX_PAD_NAND_ALE__RAWNAND_ALE         0xb0b1
+                             MX6SX_PAD_NAND_WP_B__RAWNAND_WP_B       0xb0b1
+                             MX6SX_PAD_NAND_READY_B__RAWNAND_READY_B 0xb000
+                             MX6SX_PAD_NAND_CE0_B__RAWNAND_CE0_B     0xb0b1
+                             MX6SX_PAD_NAND_RE_B__RAWNAND_RE_B       0xb0b1
+                             MX6SX_PAD_NAND_WE_B__RAWNAND_WE_B       0xb0b1
+                             MX6SX_PAD_NAND_DATA00__RAWNAND_DATA00   0xb0b1
+                             MX6SX_PAD_NAND_DATA01__RAWNAND_DATA01   0xb0b1
+                             MX6SX_PAD_NAND_DATA02__RAWNAND_DATA02   0xb0b1
+                             MX6SX_PAD_NAND_DATA03__RAWNAND_DATA03   0xb0b1
+                             MX6SX_PAD_NAND_DATA04__RAWNAND_DATA04   0xb0b1
+                             MX6SX_PAD_NAND_DATA05__RAWNAND_DATA05   0xb0b1
+                             MX6SX_PAD_NAND_DATA06__RAWNAND_DATA06   0xb0b1
+                             MX6SX_PAD_NAND_DATA07__RAWNAND_DATA07   0xb0b1
+                             >;
+                     };
+
+pinctrl_flexcan1: flexcan1grp {
+                      fsl,pins = <
+                          MX6SX_PAD_KEY_ROW2__CAN1_RX			0x1b020
+                          MX6SX_PAD_KEY_COL2__CAN1_TX			0x1b020
+                          >;
+                  };
+
+
+pinctrl_flexcan2: flexcan2grp {
+                      fsl,pins = <
+                          MX6SX_PAD_KEY_ROW3__CAN2_RX			0x1b020
+                          MX6SX_PAD_KEY_COL3__CAN2_TX			0x1b020
+                          >;
+                  };
+
+
+pinctrl_i2c1: i2c1grp {
+                  fsl,pins = <
+                      MX6SX_PAD_GPIO1_IO01__I2C1_SDA 0x4001b8b1
+                      MX6SX_PAD_GPIO1_IO00__I2C1_SCL 0x4001b8b1
+                      >;
+              };
+
+pinctrl_uart1: uart1grp {
+                   fsl,pins = <
+                       MX6SX_PAD_GPIO1_IO04__UART1_TX		0x1b0b1
+                       MX6SX_PAD_GPIO1_IO05__UART1_RX		0x1b0b1
+                       >;
+               };
+
+pinctrl_usdhc3: usdhc3grp {
+                    fsl,pins = <
+                        MX6SX_PAD_SD3_CMD__USDHC3_CMD		0x17069
+                        MX6SX_PAD_SD3_CLK__USDHC3_CLK		0x10071
+                        MX6SX_PAD_SD3_DATA0__USDHC3_DATA0	0x17069
+                        MX6SX_PAD_SD3_DATA1__USDHC3_DATA1	0x17069
+                        MX6SX_PAD_SD3_DATA2__USDHC3_DATA2	0x17069
+                        MX6SX_PAD_SD3_DATA3__USDHC3_DATA3	0x17069
+                        MX6SX_PAD_SD3_DATA4__GPIO7_IO_6		0x17059 /* CD */
+                        MX6SX_PAD_SD3_DATA5__GPIO7_IO_7		0x17059 /* WP */
+                        >;
+                };
+    };
 };
-- 
1.9.1

